Rubberduck.Parsing/ComReflection/ComField.cs

```diff
@@ -1,127 +1,23 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Diagnostics;
-using System.Linq;
-using System.Runtime.InteropServices;
+﻿using System.Diagnostics;
 using System.Runtime.InteropServices.ComTypes;
 using Rubberduck.Parsing.Symbols;
-using Rubberduck.VBEditor.Utility;
-using TYPEATTR = System.Runtime.InteropServices.ComTypes.TYPEATTR;
-using TYPEDESC = System.Runtime.InteropServices.ComTypes.TYPEDESC;
-using TYPEKIND = System.Runtime.InteropServices.ComTypes.TYPEKIND;
 using VARDESC = System.Runtime.InteropServices.ComTypes.VARDESC;
 using VARFLAGS = System.Runtime.InteropServices.ComTypes.VARFLAGS;
 
 namespace Rubberduck.Parsing.ComReflection
 {
-    [DebuggerDisplay("{Name}")]
-    public class ComField
+    [DebuggerDisplay("{" + nameof(Name) + "}")]
+    public class ComField : ComTypedElement
     {
-        public string Name { get; }
-        public int Index { get; }
-        public DeclarationType Type { get; }
-        public object DefaultValue { get; }
-        public bool IsReferenceType { get; private set; }
-
-        private string _valueType = "Object";
-        public string ValueType => IsArray ? $"{_valueType}()" : _valueType;
-
-        private Guid _enumGuid = Guid.Empty;
-        public bool IsEnumMember => !_enumGuid.Equals(Guid.Empty);
-
-        public bool IsArray { get; private set; }
+        public override DeclarationType Type { get; }
         public VARFLAGS Flags { get; }
 
-        public ComField(ITypeInfo info, string name, VARDESC varDesc, int index, DeclarationType type)
+        public ComField(ITypeInfo info, VARDESC varDesc, string name, int index, DeclarationType type) : base(info, varDesc, name, index, type == DeclarationType.Constant)
         {
-            Name = name;
-            Index = index;
             Type = type;
-
             Flags = (VARFLAGS)varDesc.wVarFlags;
```

Comment:
This seems to be the only use of the `varDesc`. Why not just pass in the `wVarFlags` instead of `VarDesc`? I think I'd prefer to avoid passing type structure across methods when it's not absolutely essential.